### PR TITLE
Unify the error handling

### PR DESCRIPTION
**Pull request by difince**
Unifies the output when an object is deleted, or fails because of an err
almost all 'remove' commands support both:
device rm device_id
device rm --name device_name

Fix: https://github.com/edgexfoundry-holding/edgex-cli/issues/64

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: cmd/device/rm/rm.go [hunk 1]
```diff
@@ -26,43 +26,39 @@ import (
...
 		Short: "Removes device by name or ID",
 		Long: `Removes a device given its name or ID. 
 You can use: '$ edgex device list' to find a device's name and ID.`,
-		Run: func(cmd *cobra.Command, args []string) {
-
-			// Checking for args
-			if len(args) == 0 {
-				fmt.Printf("Error: No device ID/Name provided.\n")
-				return
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
```
Comment (mkbhanda) [inline cmd/device/rm/rm.go:37]: space before {

File: cmd/device/rm/rm.go [hunk 2]
```diff
@@ -26,43 +26,39 @@ import (
...
 
 			ctx := context.Background()
-			err := mdc.DeleteByName(ctx, deviceID)
-			if err != nil {
-				fmt.Printf("Removed: %s\n", deviceID)
-				return
+			if name != "" {
+				err = mdc.DeleteByName(ctx, name)
+				if err == nil {
+					fmt.Printf("Removed: %s\n", name)
```
Comment (mkbhanda) [inline cmd/device/rm/rm.go:49]: Good catch!

File: cmd/deviceservice/rm/rm.go [hunk 3]
```diff
@@ -35,20 +35,15 @@ func NewCommand() *cobra.Command {
 			     "deviceservice rm --name <name>",
 		Short: "Removes device service by name or ID",
 		Long:  `Removes a device service from the core-metadata DB.`,
-		RunE: func(cmd *cobra.Command, args []string) (err error){
+		Run: func(cmd *cobra.Command, args []string) {
```
Comment (mkbhanda) [inline cmd/deviceservice/rm/rm.go:38]: thought you said RunE is the way to go .. why did you reverse here?

File: cmd/deviceservice/rm/rm.go [hunk 4]
```diff
@@ -35,20 +35,15 @@ func NewCommand() *cobra.Command {
...
 			if len(args) == 0 && name ==""{
```
Comment (mkbhanda) [inline cmd/deviceservice/rm/rm.go:39]: space before {

File: cmd/interval/rm/rm.go [hunk 5]
```diff
@@ -26,38 +27,40 @@ import (
...
-	// Create request
-	intervalID := args[0]
 	sc := scheduler.NewIntervalClient(
-		local.New(url+clients.ApiIntervalRoute),
+		local.New(config.Conf.Clients["Scheduler"].Url()+clients.ApiIntervalRoute),
 	)
-
-	err := sc.Delete(context.Background(), intervalID)
+	var deletedBy string
+	if name != ""{
```
Comment (mkbhanda) [inline cmd/interval/rm/rm.go:41]: space before {

File: cmd/notification/rm/rm.go [hunk 6]
```diff
@@ -56,7 +52,7 @@ func NewCommand() *cobra.Command {
 		Short: "Removes notification by slug or age",
 		Long:  `Removes a notification given its slug or age timestamp.`,
 		Args:  cobra.ExactValidArgs(1),
-		RunE:   removeNotificationHandler,
+		Run:   removeNotificationHandler,
```
Comment (mkbhanda) [inline cmd/notification/rm/rm.go:55]: Run instead of RunE?

File: cmd/subscription/rm/rm.go [hunk 7]
```diff
@@ -30,31 +30,26 @@ var slug string
...
 func NewCommand() *cobra.Command {
 	var cmd = &cobra.Command{
-		Use:   "rm [--slug or id]",
+		Use:   "rm [id | --slug]",
 		Example: "subscription rm <id> \n" +
 			     "subscription rm --slug <slug>",
 		Short: "Removes subscription by --slug or id.",
 		Long:  `Removes a subscription given its slug or id.`,
-		RunE:   removeSubscriptionHandler,
+		Run:   removeSubscriptionHandler,
```
Comment (mkbhanda) [inline cmd/subscription/rm/rm.go:38]: Run instead RunE?

File: pkg/request.go [hunk 8]
```diff
@@ -17,27 +17,26 @@ func Get(url string, items interface{}) (err error) {
...
 	printURL(url)
-	err :=  clients.DeleteRequest(context.Background(), "", local.New(url))
+	return clients.DeleteRequest(context.Background(), "", local.New(url))
+}
+
+func DeletePrt(url string, deletedBy string) error {
+	err := Delete(url)
 	if err != nil {
 		fmt.Printf("Error: %s\n", err.Error())
+	}else if deletedBy != ""{
```
Comment (mkbhanda) [inline pkg/request.go:34]: blank after } and before {

File: cmd/deviceservice/rm/rm.go (hunk 3 above)
Comment (difince) [reply cmd/deviceservice/rm/rm.go]: Run functions have been returned back to RunE.

**Pull request merged**